src/writers/file_writer/body_writer/mod.rs

```diff
@@ -0,0 +1,109 @@
+        }
+    }
+
+    fn get_body_string_from_endpoint(&self) -> String {
+        let mut full_output_string = String::new();
+
+        // macro header handling
+        full_output_string.push_str(&self.macro_string());
+
+        // method signature handling
+        full_output_string.push_str(&self.method_signature_string());
+
+        // method boyd handling
+        full_output_string.push_str(&self.method_body_string());
+
+        full_output_string
+    }
+
+    fn macro_string(&self) -> String {
+        let route = &self.endpoint.route;
+        format!("#[get(\"{}\")]\n", route)
+    }
+
+    fn method_signature_string(&self) -> String {
+        let mut output_string = String::new();
+
+        let mut param_string = String::new();
+
+        if let Some(query) = &self.endpoint.query_param {
+            param_string.push_str(&format!("{}: {}", query.name, query.field_type));
+        }
+
+        let fn_name = &self.endpoint.name;
+        format!(
+            "async fn {}({}) -> impl Responder {{\n",
+            fn_name, param_string
+        )
+    }
+
+    fn method_body_string(&self) -> String {
+        // setup and create the util method handler
+        let mut param_string = String::new();
+        if let Some(query) = &self.endpoint.query_param {
+            param_string.push_str(&format!("{}: {}", query.name, query.field_type));
+        }
+        let util_method = format!(
+            "let response = utils::{}_util({});",
+            &self.endpoint.name, param_string
+        );
+
+        // assemble the status code and the HttpResponse
+        let status_code = format!(
+            "let status_code = http::StatusCode::from_u16({}).unwrap();",
+            &self.endpoint.success_code
+        );
+
+        let http_response = String::from("HttpResponse::build(status_code).body(response)}");
+
+        format!("{}\n{}\n{}", util_method, status_code, http_response)
+    }
+}
+
+/// This trait is to be shared amongst all of the HTTP<verb>BodyStringBuilders, and has
+/// common util functions for them all so that unpacking calls work polymorphically
+pub trait BodyBuilder {
+    /// Dummy constructor for allowing trait usage
+    fn new(endpoint_ref: &Endpoint) -> Self;
+
+    /// Top-level function for the struct, returns final output string.
+    ///
+    /// Based entirely off of an `Endpoint` reference, decomposing it
+    /// into the final output string headed to file.
+    fn get_body_string_from_endpoint(&self) -> String;
+
+    /// Returns the actix macro string for the specific HTTP<Verb> builder
+    fn macro_string(&self) -> String;
+
+    /// Returns the actual method signature string for the actix method
+    fn method_signature_string(&self) -> String;
+
+    /// Returns the body string of the actix method.
+    ///
+    /// This method can call many helper methods as long as the result
+    /// returned is final and output-ready.
+    fn method_body_string(&self) -> String;
+}
```

Comment:
The actual code is pretty straightforward but the structure might be a little weird upon first glance so i'll break it down (although if it isnt readable without this, it's a bad smell anyway)

### the `BodyBuilder` trait
the name is funny by accident, but the idea is very solid.

All of the HTTP verbs will have slightly different implementations of the same string builder methods (macro, method sign., etc.), so adopting a trait with their different phrasing makes the system more akin to a polymorphic handler pattern vs. what would be something closer to an abstract factory (if we were to implement them all one by one and then have the user choose).

### The actual string builder methods
Writing code by formatting strings is hacky at best and a nightmare at worst, so great care went into making this system be as clean, legible, and maintainable as possible.

Basically, every major part of an `actix` endpoint method was broken down into its components and the components are handled as cleanly as possible.

This means that to extend a certain component, only a single function (per http verb struct) is edited; this is fantastic.

This also means that if we would like to add a totally _new_ component, we only need to edit the `trait` block and the compiler will force us to implement the new methods (also great).